kork/kork-docker/src/main/java/com/netflix/spinnaker/kork/docker/service/DockerRegistryClient.java

```diff
@@ -0,0 +1,787 @@
+/*
+ * Copyright 2025 Harness, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netflix.spinnaker.kork.docker.service;
+
+import com.fasterxml.jackson.databind.DeserializationFeature;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.google.gson.Gson;
+import com.netflix.spinnaker.kork.client.ServiceClientProvider;
+import com.netflix.spinnaker.kork.docker.exceptions.DockerRegistryAuthenticationException;
+import com.netflix.spinnaker.kork.docker.exceptions.DockerRegistryOperationException;
+import com.netflix.spinnaker.kork.docker.model.DockerBearerToken;
+import com.netflix.spinnaker.kork.docker.model.DockerManifest;
+import com.netflix.spinnaker.kork.docker.model.DockerRegistryCatalog;
+import com.netflix.spinnaker.kork.docker.model.DockerRegistryTags;
+import com.netflix.spinnaker.kork.retrofit.ErrorHandlingExecutorCallAdapterFactory;
+import com.netflix.spinnaker.kork.retrofit.Retrofit2SyncCall;
+import com.netflix.spinnaker.kork.retrofit.exceptions.SpinnakerHttpException;
+import com.netflix.spinnaker.kork.retrofit.exceptions.SpinnakerServerException;
+import java.io.File;
+import java.nio.file.Files;
+import java.time.Instant;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.function.Function;
+import java.util.function.Supplier;
+import lombok.Getter;
+import lombok.extern.slf4j.Slf4j;
+import okhttp3.ResponseBody;
+import retrofit2.Response;
+import retrofit2.Retrofit;
+import retrofit2.converter.jackson.JacksonConverterFactory;
+
+@Slf4j
+@Getter
+public class DockerRegistryClient {
+
+  private final String address;
+  private String email;
+  private String username;
+  private String password;
+  private String passwordCommand;
+  private File passwordFile;
+  private final File dockerconfigFile;
+  private long clientTimeoutMillis;
+  private final int paginateSize;
+  private final String catalogFile;
+  private final String repositoriesRegex;
+  private final boolean insecureRegistry;
+  private final DockerOkClientProvider okClientProvider;
+  private final ServiceClientProvider serviceClientProvider;
+
+  private DockerBearerTokenService tokenService;
+  private final RegistryService registryService;
+  private static ObjectMapper objectMapper;
+
+  static final String USER_AGENT = DockerUserAgent.getUserAgent();
```

Comment:
Should this be public?  I see used some tests, just asking...  reduces places for a docker usage agent which seems duplicated in a few spots.

Reply:
Its primarily used in the DockerRegistryClient and a couple of times in the DockerBearerTokenService but you are right that within the DockerRegistryClient i should use the USER_AGENT instead of the getter. Fixed